src/App.tsx

```diff
@@ -8,55 +8,148 @@ import { PostsList } from './components/PostsList';
+
+  const [postIdComments, setPostIdComments] = useState<number | null>(null);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [isCommentError, setIsCommentError] = useState(false);
+
+  const [isNewComment, setIsNewComment] = useState(false);
+
+  const fetchUsers = async () => {
+    try {
+      const usersFromServer = await getUsers();
+
+      setUserSelect(usersFromServer);
+    } catch (error) {
+      setIsPostError(true);
+    }
+  };
+
+  const fetchPosts = async (id: number) => {
+    try {
+      const postsFromServer = await getPosts(id);
+
+      setUserPosts(postsFromServer);
+    } catch (error) {
+      setIsPostError(true);
+    }
+  };
+
+  const fetchComments = async (id: number) => {
+    try {
+      const commentsFromServer = await getComments(id);
+
+      setPostComments(commentsFromServer);
+    } catch (error) {
+      setIsCommentError(true);
+    }
+  };
+
+  useEffect(() => {
+    fetchUsers();
+  }, []);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  userSelect={userSelect}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                  fetchPosts={fetchPosts}
+                  setIsPostsLoading={setIsPostsLoading}
+                  setPostIdComments={setPostIdComments}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
-
-                <Loader />
+                {isPostError ? (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                ) : (
+                  <>
+                    {!selectedUser && (
+                      <p data-cy="NoSelectedUser">
+                        No user selected
+                      </p>
+                    )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                    {isPostsLoading && (
+                      <Loader />
+                    )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
-
-                <PostsList />
+                    {selectedUser && !isPostsLoading && (
+                      (userPosts.length > 0) ? (
```

Comment:
It's from the checklist to another task, but here it makes sense too
![image](https://user-images.githubusercontent.com/96409563/207824606-344cacb7-9eb2-45a9-9f51-12ecb5acf108.png)